src/Sub.rei

```diff
@@ -0,0 +1,35 @@
+module type Config = {
+  type params;
+  type msg;
+
+  // State that is carried by the subscription while it is active
+  type state;
+
+  let subscriptionName: string;
+
+  let getUniqueId: params => string;
+
+  let init: (~params: params, ~dispatch: msg => unit) => state;
+  let update:
+    (~params: params, ~state: state, ~dispatch: msg => unit) => state;
+  let dispose: (~params: params, ~state: state) => unit;
+};
+
+type t('msg) = Sub_internal.t('msg);
```

Comment:
Is this equality really needed? Looks pretty awkward to have internals (by name even) exposed like that.

Reply:
Really awkward - discussed in the other comment. I'd love to get rid of this, I just haven't figured out how to do it and get the tests compiling.